lib/chef/http.rb

```diff
@@ -468,12 +468,12 @@ def version_retries
 
     # @api private
     def http_retry_delay
-      config[:http_retry_delay]
+      options[:retry_delay] || config[:http_retry_delay]
     end
 
     # @api private
     def http_retry_count
-      config[:http_retry_count]
+      options[:retries] || config[:http_retry_count]
```

Comment:
the reference to the config hash here should be moved into the constructor:

```
options[:http_retry_delay] ||= config[:http_retry_delay]
options[:http_retry_count] ||= config[:http_retry_count]
```

and i also deliberately changed the name of the options there.

Reply:
Updated this, please have a look.